containers/message-refiner/app/main.py

```
@@ -38,9 +39,86 @@ async def refine_ecr(
     data = await refiner_input.body()
 
     if sections_to_include:
-        # TODO: add logic to filter the XML message based on the sections to include
-        refined_message = Response(content=data, media_type="application/xml")
-    else:
-        refined_message = Response(content=data, media_type="application/xml")
+        sections_to_include = validate_sections_to_include(sections_to_include)
+        data = refine(data, sections_to_include)
 
+    return Response(content=data, media_type="application/xml")
+
+
+def validate_sections_to_include(sections_to_include: str | None) -> list:
+    """
+    Validates the sections to include in the refined message and returns them as a list
+    of corresponding LOINC codes.
+
+    :param sections_to_include: The sections to include in the refined message.
+    :raises ValueError: When at least one of the sections_to_inlcude is invalid.
+    :return: The sections to include in the refined message as a list of LOINC codes
+    corresponding to the sections.
+    """
+    section_LOINCs = [
+        "10164-2",  # history of present illness
+        "11369-6",  # history of immunization narrative
+        "29549-3",  # medications administered
+        "18776-5",  # plan of care note
+        "11450-4",  # problem list - reported
+        "29299-5",  # reason for visit
+        "30954-2",  # relevant diagnostic tests/laboratory data narrative
+        "29762-2",  # social history narrative
+        "46240-8",  # history of hospitalizations+outpatient visits narrative
+    ]
+
+    section_loincs = None
+    if sections_to_include:
+        sections = sections_to_include.split(",")
+        section_loincs = []
+        for section in sections:
+            if section not in section_LOINCs:
+                raise ValueError(
+                    f"{section} is invalid. Please provide a valid section."
+                )
```

I don't think we should raise an exception here because that will kill things and ultimately result in the client receiving a `500 internal server error` response. I think this function should return `section_loincs` as well as something like `error_message`. Then in `refine_ecr` if `error_message != ""` we can return a 422 and include the message we currently creating.

---

If you'd like to handle this in another PR that works for me. I don't want to keeping blocking this PR for something that can be addressed separately.